### ISSUE TITLE
Why there are same operations in both "tokenizer_suffix_function" and "model_prefix_function", are they duplicate?

### ISSUE THREAD

**Issue by nixonjin** [JoaoLages/RATransformers#4]
self.input_relation_kinds[0].cpu().detach()
self.input_relation_kinds.pop()
torch.cuda.empty_cache()

# add to global variable
 self.input_relation_kinds.append(
            torch.from_numpy(aux_input_relation_kinds).to(tokenizer_outputs['input_ids'].device)
        )

**Comment by JoaoLages** [comment]
Sorry, I did not understand. Are you mentioning [this part](https://github.com/JoaoLages/RATransformers/blob/7bf401afe8e2b12531dc75f5402c2f9ffd99aaf7/src/ratransformers/__init__.py#L60) of the code?
